### PR TITLE
[proto] Few minor code improvements

### PR DESCRIPTION
**Pull request by vfdev-5**
Description:
- call torch.log once can save few ms
- all masks should be selected with is_within_crop_area in RandomIoUCrop

### CODE REVIEW

File: torchvision/prototype/transforms/_geometry.py [hunk 1]
```diff
@@ -92,14 +92,16 @@ def __init__(
...
+        self._log_ratio = torch.log(torch.tensor(self.ratio))
+
     def _get_params(self, sample: Any) -> Dict[str, Any]:
         # vfdev-5: techically, this op can work on bboxes/segm masks only inputs without image in samples
         # What if we have multiple images/bboxes/masks of different sizes ?
         # TODO: let's support bbox or mask in samples without image
         _, height, width = query_chw(sample)
         area = height * width
 
-        log_ratio = torch.log(torch.tensor(self.ratio))
```
Comment (datumbox) [inline torchvision/prototype/transforms/_geometry.py:102]: Is it a problem that we did't send this to the right device? Also I wonder if the `log_ratio` needs to be on a tensor given it's provided as parameter on `unitform()`.

**Review by datumbox — APPROVED** [review]
LGTM, though I have a question below.

File: torchvision/prototype/transforms/_geometry.py (hunk 1 above)
Comment (datumbox) [reply torchvision/prototype/transforms/_geometry.py:102]: Sounds good. Might be good testing math as well.
